utils/getDocumentValidationFailureDetails.ts

```diff
@@ -0,0 +1,63 @@
+import type OnfidoDocumentReportBreakdown from '../types/OnfidoDocumentReportBreakdown';
+import ValidationFailure from '../types/ValidationFailure';
+
+import isFailure from './isValidationFailure';
+
+const getDocumentValidationFailureDetails = (breakdown: OnfidoDocumentReportBreakdown | null): ValidationFailure[] => {
+  const failures: ValidationFailure[] = [];
+
+  if (!breakdown) {
+    return failures;
+  }
+
+  if (breakdown.imageIntegrity) {
+    if (isFailure(breakdown.imageIntegrity.result)) {
+      if (isFailure(breakdown.imageIntegrity.breakdown?.supportedDocument?.result)) {
+        failures.push(ValidationFailure.InvalidImageIntegritySupportedDocument);
+      }
+      if (isFailure(breakdown.imageIntegrity.breakdown?.imageQuality?.result)) {
+        failures.push(ValidationFailure.InvalidImageIntegrityImageQuality);
```

Comment:
You could consider creating a backlog ticket to consider later (if we continue using Onfido and have lots of volume) to improve this further. E.g. it looks like Onfido reveals more specific validation errors that we could pass along to the user: https://documentation.onfido.com/#image-quality-reasons

Reply:
KTA-83 created